### PR TITLE
[EWL-5310] SG2 | Create "Category Page" pattern

### PR DESCRIPTION
**Pull request by csj1012**
## Ticket(s)

**Jira Ticket**
- [EWL-5310: SG2 | Create "Category Page" pattern](https://issues.ama-assn.org/browse/EWL-5310)

## Description
Adds the Category Page pattern.

## To Test
- [ ] `gulp serve`
- [ ] When I click "pages"
- [ ] Then I should see "Category"
- [ ] When I click "Category"
- [ ] Then I should see the "Masthead as Title Only"
- [ ] And I should see the "Subcategory Exploration"
- [ ] And I should see the "Category Page Hero"
- [ ] And I should see the "Tools List"
- [ ] And I should optionally see a promo in the "Promotional Real Estate" under the "Tools list"
- [ ] And I should see the "JAMA as content bottom" 
- [ ] And I should see the "Article Stub as Large for Category Page"
- [ ] And I should see the "Category Page Article Stub Four-Up"
- [ ] And I should see the "Subcategory Featured Content"
- [ ] And I should see the "Subcategory Featured Content as Carousel"
- [ ] And I should see the "Category Page Article Stub Three-Up"
- [ ] And I should optionally see a "JAMA Promo" in the Promotional Real Estate next to the "Category Page Article Stub Three-Up"
- [ ] And I should see the "Category Page Article Stub as Large Two-Up"
- [ ] And I should see the "Category Page Index"
- [ ] And I should see the "Subcategory Explorer as with Images"
- [ ] Did you test in IE 11?

## Visual Regressions

- [X] **Before proceeding:** Run visual regression tests locally to ensure new work does not introduce new bugs.

- [X] _If the new work does your work introduce visual regressions into existing work in the Style Guide (or in a Drupal 8 environment)_, either call these out here or address them before marking the PR as `ready for review`.

- [X] If you are creating or updating a pattern be sure you have created or updated the [scenario in `backstop.json`](ama-style-guide-2/styleguide/backstop.json) and have included a new or updated screenshot in [`bitmaps_reference`](ama-style-guide-2/styleguide/backstop_data/bitmaps_reference).

For more information on visual regression testing review the [How do I run tests?](https://issues.ama-assn.org:8446/confluence/pages/viewpage.action?pageId=23298568) document in Confluence.

## Relevant Screenshots/GIFs
backstop_default_category_page_0_document_4_bp-xl

## Remaining Tasks
- [ ] updating [The Glossary](https://issues.ama-assn.org:8446/confluence/display/DTD/Glossary?src=contextnavpagetreemode) 

## Additional Notes
The following are currently hard-coded due to Twig errors:
- membership
- JAMA

If I have time before code freeze I'l go back to troubleshoot.

---

[Guidelines for Contribution](CONTRIBUTING.md)
[SG2 Standards](ama-style-guide-2/docs/standards.md)

### CODE REVIEW

File: styleguide/source/_patterns/03-organisms/_category-article-three-up-jama.twig [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+  <div class="ama__category__article-stub-three-up">
+    {% for articleStub in category.articleStubsThreeUp.content %}
+      {% include "@molecules/article-stub/article-stub.twig" %}
+    {% endfor %}
+  </div>
+{% endblock %}
+
+{% block contentRight %}
+<div class="ama__category__jama--as--sidebar">
+  {# @todo - get this to work #}
```
Comment (designerneil) [inline styleguide/source/_patterns/03-organisms/_category-article-three-up-jama.twig:18]: Does this commented code need to be here?

File: styleguide/source/_patterns/03-organisms/_category-hero-tools.twig [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+{% block contentTop %}
+{% endblock %}
+{% block contentLeft %}
+  {% set categoryHero = category.categoryHero %}
+  {% include '@molecules/category-hero.twig' %}
+{% endblock %}
+{% block contentRight %}
+  {% set tools = category.tools %}
+  {% include '@organisms/tools.twig' %}
+  {# @todo replace this with actual Twig - there is an error presently. #}
```
Comment (designerneil) [inline styleguide/source/_patterns/03-organisms/_category-hero-tools.twig:11]: Can you add this in order tech debt document?

**Comment by designerneil** [comment]
this page is broken in IE 11. 
screen shot 2018-06-27 at 2 04 15 pm

**Comment by designerneil** [comment]
@chelsiejohnston I'm also getting 10 backstop failures
screen shot 2018-06-27 at 2 13 23 pm

**Comment by labbydev** [comment]
@chelsiejohnston I have created [EWL-5378](https://issues.ama-assn.org/browse/EWL-5378) to address the IE11 bug next sprint. For the time being let's fix the backstop errors and move forward.

**Review by labbydev — APPROVED** [review]

**Pull request merged**